### PR TITLE
chore: simplifies k8s client usage in Features

### PR DESCRIPTION
**Pull request by bartoszmajsak**
## Description

- simplifies interaction with cluster APIs by using basic `client.Interface`
- previously, for those CRDs we did not want to take as dependencies, `DynamicClient` was used. Now unstructured objects with GVK set instead are used with `client.Interface` instead

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by cam-garrison — APPROVED** [review]
i like it, much cleaner.

File: pkg/feature/conditions.go [hunk 1]
```diff
@@ -28,7 +28,13 @@ func WaitForPodsToBeReady(namespace string) Action {
...
 
 		return wait.PollUntilContextTimeout(context.TODO(), interval, duration, false, func(ctx context.Context) (bool, error) {
-			podList, err := f.Clientset.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{})
+			var podList corev1.PodList
+
+			err := f.Client.List(context.TODO(), &podList, client.InNamespace(namespace))
+			if err != nil {
+				return false, err
+			}
+
```
Comment (bartoszmajsak) [inline pkg/feature/conditions.go:37]:
```suggestion
```

double return (see below xD)

**Comment by zdtsw** [comment]
funny that `NetworkPolicies` and `CreateConfigMap` are not even in  use. wonder why linter was so silent

**Review by zdtsw — APPROVED** [review]

**Review by aslakknutsen — APPROVED** [review]

**Pull request merged**